composition-js/src/merging/merge.ts

```diff
@@ -2572,6 +2584,116 @@ class Merger {
       // Because we rename all root type in subgraphs to their default names, we shouldn't ever have incompatibilities here.
       assert(!isIncompatible, () => `Should not have incompatible root type for ${rootKind}`);
     }
+    this.addJoinDirectiveDirectives(sources, dest);
+  }
+
+  private normalizeSpecIdentityURL(identityURL: string): string {
```

Comment:
consider using FeatureURL.parse as I think it does what you need.

Reply:
https://github.com/apollographql/federation/pull/2894/commits/d803501c4085eddc0717881d6e90a798e034a0b2